phoenicis-apps/src/main/java/org/phoenicis/apps/GitApplicationsSource.java

```diff
@@ -18,45 +18,109 @@
+import org.phoenicis.apps.dto.CategoryDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 class GitApplicationsSource implements ApplicationsSource {
-    private final String gitRepositoryURL;
-    private final LocalApplicationsSource.Factory localAppsSourceFactory;
-    private List<CategoryDTO> cache;
-
-    GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
-        this.gitRepositoryURL = gitRepositoryURL;
-        this.localAppsSourceFactory = localAppsSourceFactory;
-    }
-
-    @Override
-    public synchronized List<CategoryDTO> fetchInstallableApplications() {
-        if (cache != null) {
-            return cache;
-        }
-
-        try {
-            final File gitTmp = Files.createTempDirectory("git").toFile();
-            gitTmp.deleteOnExit();
-            new ProcessBuilder(Arrays.asList("git", "clone", gitRepositoryURL, gitTmp.getAbsolutePath()))
-                    .inheritIO()
-                    .start()
-                    .waitFor();
-
-            cache = localAppsSourceFactory.createInstance(gitTmp.getAbsolutePath()).fetchInstallableApplications();
-            return cache;
-        } catch (IOException | InterruptedException e) {
-            return Collections.emptyList();
-        }
-    }
+	private final static Logger LOGGER = LoggerFactory.getLogger(GitApplicationsSource.class);
+
+	private final String gitRepositoryURL;
+	private final LocalApplicationsSource.Factory localAppsSourceFactory;
+	private List<CategoryDTO> cache;
+
+	private static final String cacheDirectoryPath = System.getProperty("java.io.tmpdir");
+
+	GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
+		this.gitRepositoryURL = gitRepositoryURL;
+		this.localAppsSourceFactory = localAppsSourceFactory;
+	}
+
+	@Override
+	public synchronized List<CategoryDTO> fetchInstallableApplications() {
+		if (cache != null) {
+			return cache;
+		}
+
+		File gitRepositoryLocation = new File(
+				GitApplicationsSource.cacheDirectoryPath + "/git" + this.gitRepositoryURL.hashCode());
+
+		LOGGER.info(String.format("Git-repository '%s' will be saved in '%s'", this.gitRepositoryURL,
+				gitRepositoryLocation.getAbsolutePath()));
+
+		/*
+		 * This value is true if a folder for the git repository exists,
+		 * otherwise it is false
+		 */
+		boolean folderExists = true;
+
+		// check that the repository folder exists
+		if (!gitRepositoryLocation.exists()) {
+			folderExists = false;
+
+			LOGGER.info(String.format("Creating new folder '%s' for git-repository '%s'",
+					gitRepositoryLocation.getAbsolutePath(), this.gitRepositoryURL));
+
+			if (!gitRepositoryLocation.mkdir()) {
+				LOGGER.error(String.format("Couldn't create folder for git repository '%s' at '%s'",
+						this.gitRepositoryURL, gitRepositoryLocation.getAbsolutePath()));
+
+				return Collections.emptyList();
```

Comment:
The current idea is, that if the folder couldn't be created, simply return at this point an empty collection without setting the cache, because:

- This prevents the potentially git initialization code from being called (the code in the try/catch block).
- Because cache isn't set here it makes it possible for POL 5 to retry the generation of the directory at a later time (more detailed, after the method is called a second time).